authors/apps/authentication/views.py

```diff
@@ -73,3 +78,68 @@ def update(self, request, *args, **kwargs):
 
         return Response(serializer.data, status=status.HTTP_200_OK)
 
+
+class SocialSignUp(CreateAPIView):
+    renderer_classes = (UserJSONRenderer,)
+    serializer_class = SocialSignUpSerializer
+    #allow everyone to view without having to be authenticated
+    permission_classes = (AllowAny,)
+
+    def create(self, request, *args, **kwargs):
+        """
+        Override `create` instead of `perform_create` to access request
+        request is necessary for `load_strategy`
+        """
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        provider = serializer.data['provider']
+
+        # If this request was made with an authenticated user, try to associate this social 
+        # account with it
+        authed_user = request.user if not request.user.is_anonymous else None
+        strategy = load_strategy(request)
+        #Get the backend that is associated with the user provider i.e google,twitter and facebook
+        backend = load_backend(strategy=strategy, name=provider, redirect_uri=None)
+
+
+        if isinstance(backend, BaseOAuth1):
+           #cater for services that use OAuth1, an example is twitter
+            token = {
+                'oauth_token': serializer.data['access_token'],
+                'oauth_token_secret': request.data['access_token_secret'],
+            }
+            print(token)
+        elif isinstance(backend, BaseOAuth2):
+            #we just need to pass access_token for OAuth2
+            token = serializer.data['access_token']
+            print(token)
```

Comment:
This one too.

Reply:
I will remove this one too @codingedward